crates/sui-framework/packages/sui-framework/sources/test/test_scenario.move

```diff
@@ -103,6 +103,146 @@ public fun begin(sender: address): Scenario {
     }
 }
 
+/// Update the `Scenario` context to use `ids_created` as the number of IDs created
+public fun with_ids_created(
```

Comment:
My main concern about this setup is that it allows changing these values "mid" transaction, when they feel like something that should be set only "between" transactions. 
There is nothing inherently wrong I suppose about setting them mid "transaction", but they do let someone create a test environment that doesn't quite represent reality. This is the idea behind functions like `next_epoch` forcibly calling `next_tx`. 

Obviously test_scenario isn't perfect w.r.t. restoring objects (because we really can't due to wrapping), we do however enforce a good deal of checks at these transaction boundaries, and that might be a nice design principle to continue to uphold.